### ISSUE TITLE
Shards install - Out of memory

### ISSUE THREAD

**Issue by Barogthor** [crystal-lang/shards#287]
When trying to install the shards I get an unexpected exception and I don't know how to solve it even after searching.
I use shards 0.9.0 and once the install reach a certain point, there seems to be an infinite loop increasing memory consumption until it crash (several minutes later).

(I had 1Go used and 0 from swap before starting shards)
![2](https://user-images.githubusercontent.com/25849838/60821493-e679e480-a1a3-11e9-973a-506d0c89483a.PNG)
![3](https://user-images.githubusercontent.com/25849838/60821502-ebd72f00-a1a3-11e9-936a-2036b3d0f43a.PNG)

**Comment by j8r** [comment]
Duplicate of https://github.com/crystal-lang/shards/issues/282
I think this issue is important, I don't see Crystal 0.30.0 bundled with shards 0.9.0. This will cause lots of issues like this one.

**Comment by ysbaddaden** [comment]
Closed as duplicate. I wish I could fix this, but it's very hard.
